app/controllers/AmendCheckYourAnswersController.scala

```diff
@@ -59,6 +59,12 @@ class AmendCheckYourAnswersController @Inject()(
 
   def onSubmit: Action[AnyContent] = (identify andThen getData andThen requireData).async {
     implicit request =>
+      val checkYourAnswersHelper = new CheckYourAnswersHelper(request.userAnswers)
```

Comment:
Hi @somaraja could we have. chat about this tomorrow. I think it may be best to have this logic in the buildValidAmendRequest.